src/ucp/tag/tag_match.inl

```diff
@@ -238,62 +239,64 @@ ucp_request_recv_offload_data(ucp_request_t *req, const void *data,
      * adding length of already received data.
      * NOTE: total length of unexpected eager offload message is not known
      * until last fragment arrives. */
-    offset = req->recv.offset;
+    offset = req->recv.dt_iter.offset;
 
     if (ucs_unlikely(req->status != UCS_OK)) {
         goto out;
     }
 
-    if (ucs_unlikely(req->recv.length < (length + offset))) {
+    if (ucs_unlikely(req->recv.dt_iter.length < (length + offset))) {
         /* We have to release non-contig buffer only in case of
          * this is not the first segment and the datatype is
          * non-contig */
-        if ((offset != 0) && !UCP_DT_IS_CONTIG(req->recv.datatype)) {
+        if ((offset != 0) &&
+            (req->recv.dt_iter.dt_class != UCP_DATATYPE_CONTIG)) {
             ucp_tag_recv_request_release_non_contig_buffer(req);
         }
         req->status = ucp_request_recv_msg_truncated(req, length, offset);
         goto out;
     }
 
-    if (UCP_DT_IS_CONTIG(req->recv.datatype)) {
-        ucp_request_unpack_contig(req,
-                                  UCS_PTR_BYTE_OFFSET(req->recv.buffer, offset),
-                                  data, length);
-    } else {
+    if (req->recv.dt_iter.dt_class == UCP_DATATYPE_CONTIG) {
```

Comment:
minor: maybe it is worth to have a macro/"inline func" for this

Reply:
not sure since it's simple comparison (not bitmap check as with datatype), but in any case if introducing such change it has to be another PR since it will change many places, WDYT?